lib/pusher.js

```diff
@@ -229,6 +231,14 @@ Pusher.prototype.createSignedQueryString = function(options) {
   return requests.createSignedQueryString(this.config.token, options);
 };
 
+Pusher.prototype.channelSharedSecret = function(channel) {
+  return crypto.createHash('sha256').update(channel + this.config.encryptionMasterKey).digest();
+}
+
+Pusher.prototype.isEncryptedChannel = function(channel) {
+  return channel.startsWith("private-encrypted-");
+}
```

Comment:
I don't think this should be attached to the Pusher prototype. I think it could be a plain function in `util.js`. It would help avoid the pusher object being passed into function it doesn't need to be, e.g. `getSocketSignature`.

Reply:
Agree